File: src/main/java/me/nylestroke/nylemod/command/RandomTPCommand.java
package me.nylestroke.nylemod.command;

import com.mojang.brigadier.CommandDispatcher;
import net.minecraft.server.command.CommandManager;
import net.minecraft.server.command.ServerCommandSource;
import net.minecraft.server.network.ServerPlayerEntity;
import net.minecraft.text.LiteralText;
import net.minecraft.world.World;

import java.util.Random;

public class RandomTPCommand {

    public static void register(CommandDispatcher<ServerCommandSource> dispatcher, boolean dedicated) {
        dispatcher.register(CommandManager.literal("randomtp").executes((context -> {
                         ServerCommandSource source = context.getSource();
                         ServerPlayerEntity player = source.getPlayer();
                         World world = source.getWorld();

                         double ranX = new Random().nextDouble(player.getX(), player.getX() + 500);
                         double ranY = player.getY();
                         double ranZ = new Random().nextDouble(player.getZ(), player.getZ() + 500);

                        player.requestTeleport(ranX, ranY, ranZ);

                         source.sendFeedback(new LiteralText("§e[RandomTP]§r Location Coordinates: §8(" +
                                 Math.round(ranX) + ", " + Math.round(ranY) + ", " + Math.round(ranZ) + ")§r"), false);
                        return 1;

                })));
    }
}

File: src/main/kotlin/me/nylestroke/nylemod/command/RandomTPCommand.kt
package me.nylestroke.nylemod.command

import com.mojang.brigadier.Command
import com.mojang.brigadier.CommandDispatcher
import com.mojang.brigadier.context.CommandContext
import net.minecraft.command.CommandRegistryAccess
import net.minecraft.server.command.CommandManager
import net.minecraft.server.command.CommandManager.RegistrationEnvironment
import net.minecraft.server.command.ServerCommandSource
import net.minecraft.server.network.ServerPlayerEntity
import net.minecraft.text.Text
import net.minecraft.world.World
import java.util.*
import kotlin.math.roundToInt

object RandomTPCommand {
    fun register(serverCommandSourceCommandDispatcher: CommandDispatcher<ServerCommandSource>,
                 commandRegistryAccess: CommandRegistryAccess,
                 registrationEnvironment: RegistrationEnvironment) {
        serverCommandSourceCommandDispatcher.register(CommandManager.literal("randomtp").executes(Command<ServerCommandSource> { context: CommandContext<ServerCommandSource> ->
            val source: ServerCommandSource = context.source
            val player: ServerPlayerEntity? = source.player
            val world: World = source.world
            val ranX = Random().nextDouble(player!!.x, player!!.x + 500)
            val ranY: Double = player!!.y
            val ranZ = Random().nextDouble(player.z, player.z + 500)
            player.requestTeleport(ranX, ranY, ranZ)
            source.sendFeedback(Text.literal("§e[RandomTP]§r Location Coordinates: §8(" +
                    ranX.roundToInt() + ", " + ranY.roundToInt() + ", " + ranZ.roundToInt() + ")§r"), false)
            1
        }))
    }
}